File: Harvest/Models/Value.swift
import Foundation

struct Value: Codable, Identifiable, Sendable {
    let id: String
    let name: String
    let category: String
    let displayOrder: Int?

    enum CodingKeys: String, CodingKey {
        case id, name, category
        case displayOrder = "display_order"
    }
}

struct UserValue: Codable, Identifiable, Sendable {
    let id: String
    let userId: String
    let valueId: String
    let ranking: Int?

    enum CodingKeys: String, CodingKey {
        case id, ranking
        case userId = "user_id"
        case valueId = "value_id"
    }
}

File: android/app/src/main/java/com/harvestglass/harvest/data/model/Value.kt
package com.harvestglass.harvest.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Mirrors Harvest/Models/Value.swift. */
@Serializable
data class Value(
    val id: String,
    val name: String,
    val category: String,
    @SerialName("display_order") val displayOrder: Int? = null
)

@Serializable
data class UserValue(
    val id: String,
    @SerialName("user_id") val userId: String,
    @SerialName("value_id") val valueId: String,
    val ranking: Int? = null
)
